pgjdbc/src/test/java/org/postgresql/test/jdbc42/SetObject310Test.java

```diff
@@ -279,6 +279,27 @@ private void offsetTimestamps(ZoneId dataZone, LocalDateTime localDateTime, Stri
     }
   }
 
+  @Test
+  public void testLocalDateTimeRounding() throws SQLException {
+    LocalDateTime dateTime = LocalDateTime.parse("2018-06-03T23:59:59.999999500");
+    localTimestamps(ZoneOffset.UTC, dateTime, "2018-06-04 00:00:00");
+  }
+
+  @Test
+  public void testTimeStampRounding() throws SQLException {
+    LocalTime time = LocalTime.parse("23:59:59.999999500");
+    Time actual = insertThenReadWithoutType(time, "time_without_time_zone_column", Time.class);
+    assertEquals(Time.valueOf("24:00:00"), actual);
+  }
+
+  @Test
+  public void testTimeStampRoundingWithType() throws SQLException {
+    LocalTime time = LocalTime.parse("23:59:59.999999500");
+    Time actual =
+        insertThenReadWithType(time, Types.TIME, "time_without_time_zone_column", Time.class);
+    assertEquals(Time.valueOf("24:00:00"), actual);
+  }
+
```

Comment:
Please consider writing a test for year change, like for '2018-12-31 23:59:59.999999500' to prevent potential bugs after refactorings in the future.